test/distrib/cpp/run_distrib_test.sh

```diff
@@ -30,9 +30,14 @@
 
 set -ex
 
-git clone --recursive $EXTERNAL_GIT_ROOT
+git clone $EXTERNAL_GIT_ROOT
 cd grpc
 
+# clone gRPC submodules, use data from locally cloned submodules where possible
+git submodule update --init --recursive --reference $EXTERNAL_GIT_SUBMODULES_ROOT
+# if local gRPC submodules clone fails, clone via network
+git submodule update --init --recursive
```

Comment:
Actually, I don't see why the previous command could fail (git should fallback by itself, right?).